packages/shared/src/components/post/PostEngagements.tsx

```diff
@@ -0,0 +1,164 @@
+import dynamic from 'next/dynamic';
+import React, { ReactElement, useEffect, useState } from 'react';
+import { useAuthContext } from '../../contexts/AuthContext';
+import { Comment } from '../../graphql/comments';
+import { Post } from '../../graphql/posts';
+import { PostOrigin } from '../../hooks/analytics/useAnalyticsContextData';
+import { usePostComment } from '../../hooks/usePostComment';
+import { useShareComment } from '../../hooks/useShareComment';
+import { useUpvoteQuery } from '../../hooks/useUpvoteQuery';
+import { AuthTriggers } from '../../lib/auth';
+import { AuthorOnboarding } from './AuthorOnboarding';
+import { NewComment } from './NewComment';
+import { PostActions, ShareBookmarkProps } from './PostActions';
+import { PostComments } from './PostComments';
+import { PostUpvotesCommentsCount } from './PostUpvotesCommentsCount';
+
+const UpvotedPopupModal = dynamic(
```

Comment:
Maybe we can use the lazymodal for these?

Reply:
I was able to migrate the `UpvotedPopupModal` but for the others, I tried and it produced dependency cycle that is not easy to fix, I think we should create a ticket dedicated to migrating these legacy implementations. WDYT?